Employment for towns by broad industry groups, Business Register and Employment Survey (BRES), 2018

santhosh-thangavel opened the pull request:
1. Completed Stage -1 transform
2. Completed post processing
3. Added cubes class
4. Added TransformTrace

santhosh-thangavel committed:
1. Stage -1 transform is completed
2. Working on stage - 2 transform

santhosh-thangavel committed:
1. stage -1 completed
2. Post processing done
3. cubes class and Tracetransform is yet to be added

santhosh-thangavel committed:
1. cubes class added
2. Tracetransform added

RedWalters commented on datasets/ONS-Employment-for-towns-by-broad-industry-groups-Business-Register-and-Employment-Survey-BRES/main.py:
```diff
@@ -0,0 +1,91 @@
+
+df['DATAMARKER'].unique()
+
+df['Industry'] = df['Industry'].map(lambda x: 'mining-quarrying-utilities' 
+                                    if (x == 'Mining, quarrying & utilities (B,D and E)') 
+                                    else 'arts-entertainment-recreation-other-services' 
+                                    if (x == 'Arts, entertainment, recreation & other services (R,S,T and U)')
+                                    else 'total' if (x == 'Total') else x)
+trace.Industry("If multiple letters are referenced in the label, label is pathified manually")
+
+df['Industry'] = df['Industry'].map(lambda x: x[-2] if x[-1]== ')' else x )
+trace.Industry("If a single letter is referenced in the label, letter is returned")
+
+df['Industry'] = df['Industry'].map(lambda x: "http://gss-data.org.uk/def/trade/concept/standard-industrial-classification-2007/"+x 
+                                    if len(x) == 1 
+                                    else "http://gss-data.org.uk/data/gss_data/trade/ons-employment-for-towns-by-broad-industry-groups#concept/industry/"+x
+                                   if len(x) == 5 
+                                    else "http://gss-data.org.uk/data/gss_data/trade/ons-employment-for-towns-by-broad-industry-groups#concept/industry"+x)
+trace.Industry("prefix are added to values")
+
+df
```
```suggestion
df['Industry'] = df['Industry'].map(lambda x: 'mining-quarrying-utilities'
                                    if (x == 'Mining, quarrying & utilities (B,D and E)')
                                    else 'arts-entertainment-recreation-other-services'
                                    if (x == 'Arts, entertainment, recreation & other services (R,S,T and U)')
                                    else 'wholesale'
                                    if (x == 'Wholesale (Part G)')
                                    else 'retail'
                                    if (x == 'Retail (Part G)')
                                    else 'motor-trades'
                                    if (x == 'Motor trades (Part G)')
                                    else 'total' if (x == 'Total') else x)
trace.Industry("If multiple letters are referenced in the label, label is pathified manually")

df['Industry'] = df['Industry'].map(lambda x: x[-2] if x[-1]== ')' else x )
trace.Industry("If a single letter is referenced in the label, letter is returned")

df['Industry'] = df['Industry'].map(lambda x: "http://gss-data.org.uk/def/trade/concept/standard-industrial-classification-2007/"+x
                                    if len(x) == 1
                                    else "http://gss-data.org.uk/data/gss_data/trade/ons-employment-for-towns-by-broad-industry-groups#concept/industry/"+x
                                   if len(x) == 5
                                    else "http://gss-data.org.uk/data/gss_data/trade/ons-employment-for-towns-by-broad-industry-groups#concept/industry/"+x)
trace.Industry("prefix are added to values")

df = df.rename(columns={'OBS' : 'Value', 'DATAMARKER' : 'Marker'})

df['Measure Type'] = 'count'
df['Unit'] = 'person'

df = df.replace({'Marker' : {'!' : 'suppressed'}})

df['Period'] = df.apply(lambda x: 'year/' + str(x['Period']), axis = 1)

df = df[['Period', 'Region', 'Industry', 'Value', 'Marker', 'Measure Type', 'Unit']]

df
```

santhosh-thangavel replied:
1.Three more values in 'Industry' column is manually pathified
2. OBS and DATAMARKER columns are renamed
3. Value in marker column is replaced
4. Period column is formatted

santhosh-thangavel committed:
1. Three more values in 'Industry' column is manually pathified
2. OBS and DATAMARKER columns are renamed
3. Value in marker column is replaced
4. Period is formatted
